### PR TITLE
Implement a little library that contains general utilities

### PR DESCRIPTION
**Pull request by JonathanDUFOUR**
This pull request is related to the issue #6. Every requested function has been implemented and tested. The only difference is their name. Here is the list of the actual names for the functions of the library:
- memCmp
- memCpy
- memMov
- memSet
- strCmp
- strCpy
- strLen

### CODE REVIEW

**Review by nils-mathieu — APPROVED** [review]
lgtm

File: lib/Makefile [hunk 1]
```diff
@@ -16,65 +16,124 @@
...
 clean:
-	@${RM} $(OBJ_DIR)
-	@echo "Cleaned object files."
+	@${RM} $(OBJ_DIR) $(NAME) $(TEST)
+	@echo "Removed object files, $(NAME), and $(TEST)."
 
 .PHONY: fclean
 fclean: clean
-	@${RM} $(NAME)
-	@echo "Removed $(NAME)."
```
Comment (eirys) [inline lib/Makefile:77]: No

File: lib/Makefile [hunk 2]
```diff
@@ -16,65 +16,124 @@
...
-	@$(CXX) $(CFLAGS) -c $< -o $@
+	@$(CXX) $(CXXFLAGS) $< $(OUTPUT_OPTION)
+
 
 .PHONY: clean
 clean:
-	@${RM} $(OBJ_DIR)
-	@echo "Cleaned object files."
+	@${RM} $(OBJ_DIR) $(NAME) $(TEST)
+	@echo "Removed object files, $(NAME), and $(TEST)."
```
Comment (eirys) [inline lib/Makefile:130]: No

File: lib/Makefile [hunk 3]
```diff
@@ -16,65 +16,124 @@
...
 
 .PHONY: fclean
 fclean: clean
-	@${RM} $(NAME)
-	@echo "Removed $(NAME)."
 
 .PHONY: re
-re: fclean all
\ No newline at end of file
+re: clean all
```
Comment (eirys) [inline lib/Makefile:136]: No

File: lib/Makefile [hunk 4]
```diff
@@ -16,65 +16,124 @@
...
+
+.PHONY: fre
+fre: fclean all
```
Comment (eirys) [inline lib/Makefile:139]: No

File: lib/Makefile
Comment (eirys) [inline lib/Makefile:1]: Redo makefile properly

File: Makefile
Comment (eirys) [inline Makefile:1]: revert unwanted changes

File: lib/include/types.h
Comment (eirys) [inline lib/include/types.h:1]: using not typedef

File: lib/src/test/main.cpp
Comment (eirys) [inline lib/src/test/main.cpp:1]: rename to test or sth more explicit

File: lib/include/types.h [hunk 5]
```diff
@@ -1,26 +1,31 @@
...
 /* ************************************************************************** */
 
 #pragma once
 
-using u8 = unsigned char;
-using u16 = unsigned short;
-using u32 = unsigned int;
-using u64 = unsigned long long;
+#include <cstdint>
+#include <unistd.h>
```
Comment (eirys) [inline lib/include/types.h:16]: allowed?? and why??

File: lib/include/types.h [hunk 6]
```diff
@@ -1,26 +1,31 @@
...
 
-using i8 = char;
-using i16 = short;
-using i32 = int;
-using i64 = long long;
+typedef uint8_t u8;
+typedef uint16_t u16;
+typedef uint32_t u32;
+typedef uint64_t u64;
+typedef size_t usize;
```
Comment (eirys) [inline lib/include/types.h:22]: discard and use u32

**Review by eirys — CHANGES_REQUESTED** [review]
lots of reverting

File: src/ui/terminal.h [hunk 7]
```diff
@@ -54,7 +54,7 @@ class Terminal final {
     /* ---------------------------------------- */
 
     inline
-    void putString(const i8* str) {
+    void putString(const char* str) {
```
Comment (eirys) [inline src/ui/terminal.h:57]: ?????

**Review by eirys — COMMENTED** [review]
requires manual merging lol! pls pull

**Pull request merged**